Maintained by Andrew <andrew@webupd8.org>
diff --git a/src/nemo-connect-server-dialog.c b/src/nemo-connect-server-dialog.c
index 38ae31d..4841357 100644
--- a/src/nemo-connect-server-dialog.c
+++ b/src/nemo-connect-server-dialog.c
@@ -694,9 +694,14 @@ connect_dialog_response_cb (NemoConnectServerDialog *dialog,
 		break;
 	case GTK_RESPONSE_HELP :
 		error = NULL;
-		gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
-			      "help:gnome-help/nemo-connect",
-			      gtk_get_current_event_time (), &error);
+		if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+				"help:ubuntu-help/nemo-connect",
+				gtk_get_current_event_time (), &error);
+		else
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+				"help:gnome-help/nemo-connect",
+				gtk_get_current_event_time (), &error);
 		if (error) {
 			eel_show_error_dialog (_("There was an error displaying help."), error->message,
 					       GTK_WINDOW (dialog));
diff --git a/src/nemo-properties-window.c b/src/nemo-properties-window.c
index f625d70..bf32eb2 100644
--- a/src/nemo-properties-window.c
+++ b/src/nemo-properties-window.c
@@ -3037,7 +3037,11 @@
 	GtkWidget *icon_pixmap_widget;
 	GtkWidget *volume_usage;
 	GtkWidget *hbox, *vbox;
-	hbox = create_page_with_hbox (window->details->stack, "basic", _("Basic"),
+	if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+		hbox = create_page_with_hbox (window->details->stack, "basic", _("Basic"),
+				      "help:ubuntu-help/nemo-file-properties-basic");
+	else
+		hbox = create_page_with_hbox (window->details->stack, "basic", _("Basic"),
 				      "help:gnome-help/nemo-file-properties-basic");
 	
 	/* Icon pixmap */
@@ -4484,8 +4488,12 @@
 	GtkGrid *page_grid;
 	char *file_name, *prompt_text;
 	GList *file_list;
-
-	vbox = create_page_with_vbox (window->details->stack, "permissions",
+	if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+		vbox = create_page_with_vbox (window->details->stack, "permissions",
+				      _("Permissions"),
+				      "help:ubuntu-help/nemo-file-properties-permissions");
+	else
+		vbox = create_page_with_vbox (window->details->stack, "permissions",
 				      _("Permissions"),
 				      "help:gnome-help/nemo-file-properties-permissions");
 
@@ -4786,7 +4794,10 @@
 	gtk_widget_show (vbox);
 	g_free (mime_type);
 
-	g_object_set_data_full (G_OBJECT (vbox), "help-uri", g_strdup ("help:gnome-help/files-open"), g_free);
+	if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+		g_object_set_data_full (G_OBJECT (vbox), "help-uri", g_strdup ("help:ubuntu-help/files-open"), g_free);
+	else
+		g_object_set_data_full (G_OBJECT (vbox), "help-uri", g_strdup ("help:gnome-help/files-open"), g_free);
 	gtk_container_set_border_width (GTK_CONTAINER (vbox), STACK_INNER_BORDER);
 	gtk_stack_add_titled (window->details->stack,
 				  vbox, "open_with", _("Open With"));
@@ -5172,10 +5183,16 @@
 	case GTK_RESPONSE_HELP:
 		curpage = gtk_stack_get_visible_child (window->details->stack);
 		helpuri = g_object_get_data (G_OBJECT (curpage), "help-uri");
-		gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
-			      helpuri ? helpuri : "help:gnome-help/files",
-			      gtk_get_current_event_time (),
-			      &error);
+		if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+				helpuri ? helpuri : "help:ubuntu-help/files",
+				gtk_get_current_event_time (),
+				&error);
+		else
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+				helpuri ? helpuri : "help:gnome-help/files",
+				gtk_get_current_event_time (),
+				&error);
 		if (error != NULL) {
 			eel_show_error_dialog (_("There was an error displaying help."), error->message,
 					       GTK_WINDOW (dialog));
diff --git a/src/nemo-view.c b/src/nemo-view.c
index ab03b44..a5b3f64 100644
--- a/src/nemo-view.c
+++ b/src/nemo-view.c
@@ -1515,7 +1515,12 @@ pattern_select_response_cb (GtkWidget *dialog, int response, gpointer user_data)
 		break;
 	case GTK_RESPONSE_HELP :
 		error = NULL;
-		gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+		if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
+				"help:ubuntu-help/files-select",
+				gtk_get_current_event_time (), &error);
+		else
+			gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (dialog)),
 			      "help:gnome-help/files-select",
 			      gtk_get_current_event_time (), &error);
 		if (error) {
diff --git a/src/nemo-window-menus.c b/src/nemo-window-menus.c
index 8f65638..bd76ff0 100644
--- a/src/nemo-window-menus.c
+++ b/src/nemo-window-menus.c
@@ -343,30 +343,38 @@ action_nemo_manual_callback (GtkAction *action,
 	GError *error;
 	GtkWidget *dialog;
 	const char* helpuri;
+	char* helpprefix;
 	const char* name = gtk_action_get_name (action);
 
 	error = NULL;
 	window = NEMO_WINDOW (user_data);
+	
+	if (!g_strcmp0(g_getenv("XDG_CURRENT_DESKTOP"), "Unity"))
+		helpprefix = "ubuntu-help";
+	else
+		helpprefix = "gnome-help";
 
 	if (g_str_equal (name, "NemoHelpSearch")) {
-		helpuri = "help:gnome-help/files-search";
+		 helpuri = g_strconcat ("help:", helpprefix, "/files-search", NULL);
 	} else if (g_str_equal (name,"NemoHelpSort")) {
-		helpuri = "help:gnome-help/files-sort";
+		helpuri = g_strconcat ("help:", helpprefix, "/files-sort", NULL);
 	} else if (g_str_equal (name, "NemoHelpLost")) {
-		helpuri = "help:gnome-help/files-lost";
+		helpuri = g_strconcat ("help:", helpprefix, "/files-lost", NULL);
 	} else if (g_str_equal (name, "NemoHelpShare")) {
-		helpuri = "help:gnome-help/files-share";
+		 helpuri = g_strconcat ("help:", helpprefix, "/files-share", NULL);
 	} else {
-		helpuri = "help:gnome-help/files";
+		helpuri = g_strconcat ("help:", helpprefix, "/files", NULL);
 	}
 
 	if (NEMO_IS_DESKTOP_WINDOW (window)) {
-		nemo_launch_application_from_command (gtk_window_get_screen (GTK_WINDOW (window)), "gnome-help", FALSE, NULL);
+		nemo_launch_application_from_command (gtk_window_get_screen (GTK_WINDOW (window)), helpprefix, FALSE, NULL);
 	} else {
 		gtk_show_uri (gtk_window_get_screen (GTK_WINDOW (window)),
 			      helpuri,
 			      gtk_get_current_event_time (), &error);
 	}
+	
+	g_free(helpuri);
 
 	if (error) {
 		dialog = gtk_message_dialog_new (GTK_WINDOW (window),
