Maintained by Andrew <andrew@webupd8.org>
diff --git a/libnemo-private/nemo-thumbnails.c b/libnemo-private/nemo-thumbnails.c
index 08876b5..2605258 100644
--- a/libnemo-private/nemo-thumbnails.c
+++ b/libnemo-private/nemo-thumbnails.c
@@ -44,7 +44,7 @@
 #include <sys/wait.h>
 #include <unistd.h>
 #include <signal.h>
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 #include "nemo-file-private.h"

diff --git a/src/nemo-properties-window.c b/src/nemo-properties-window.c
index 08876b1..2605259 100644
--- a/src/nemo-properties-window.c
+++ b/src/nemo-properties-window.c
@@ -38,7 +38,7 @@
 #include <cairo.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 #include <eel/eel-accessibility.h>
 #include <eel/eel-glib-extensions.h>
--- a/src/nemo-main-application.c
+++ b/src/nemo-main-application.c
@@ -74,7 +74,7 @@
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
 
-#include <libcinnamon-desktop/gnome-desktop-thumbnail.h>
+#include <libgnome-desktop/gnome-desktop-thumbnail.h>
 
 /* Keep window from shrinking down ridiculously small; numbers are somewhat arbitrary */
 #define APPLICATION_WINDOW_MIN_WIDTH	300

