Maintained by Andrew <andrew@webupd8.org>
Index: nemo-1.7.6/data/nemo.desktop.in.in
===================================================================
--- nemo-1.7.6.orig/data/nemo.desktop.in.in	2013-05-04 10:02:46.885816391 +1000
+++ nemo-1.7.6/data/nemo.desktop.in.in	2013-05-04 10:02:46.882483057 +1000
@@ -106,7 +106,7 @@
 Comment[zh_CN]=访问和组织文件
 Comment[zh_HK]=存取與組織檔案
 Exec=nemo %U
-Icon=folder
+Icon=system-file-manager
 Terminal=false
 Type=Application
 StartupNotify=false
