Maintained by Andrew <andrew@webupd8.org>
diff --git a/data/change-background.nemo_action.in b/data/change-background.nemo_action.in
index b903c3b..9c91fb1 100644
--- a/data/change-background.nemo_action.in
+++ b/data/change-background.nemo_action.in
@@ -4,7 +4,7 @@
 
 Comment=Change Desktop _Background
 
-Exec=cinnamon-settings backgrounds
+Exec=gnome-control-center background
 
 Selection=None
 
@@ -12,7 +12,7 @@
 
 Extensions=any;
 
-Dependencies=cinnamon-settings;
+Dependencies=gnome-control-center;
 
-Conditions=desktop;dbus org.Cinnamon;
+Conditions=desktop;

diff --git a/data/set-as-background.nemo_action.in b/data/set-as-background.nemo_action.in
index b903c6b..9c91fb8 100644
--- a/data/set-as-background.nemo_action.in
+++ b/data/set-as-background.nemo_action.in
@@ -4,10 +4,8 @@
 
 Comment=Set the selected image as your desktop wallpaper
 
-Exec=gsettings set org.cinnamon.desktop.background picture-uri "%U"
+Exec=gsettings set org.gnome.desktop.background picture-uri "%U"
 
 Selection=s
 
 Mimetypes=image/*;
-
-Conditions=dbus org.Cinnamon;
